esr/thing_test.go

```diff
@@ -0,0 +1,729 @@
+package main
+
+import (
+	"context"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/sdoque/mbaigo/components"
+	"github.com/sdoque/mbaigo/forms"
+	"github.com/sdoque/mbaigo/usecases"
+)
+
+// ----------------------------------------------------- //
+// Help functions and structs to tests initTemplate()
+// ----------------------------------------------------- //
+
+func TestInitTemplate(t *testing.T) {
+	expectedServices := []string{"register", "query", "unregister", "status"}
+
+	ua := initTemplate()
+	services := ua.GetServices()
+
+	// Check if expected name and services are present
+	if ua.GetName() != "registry" {
+		t.Errorf("Name mismatch expected 'registry', got: %s", ua.GetName())
+	}
+
+	for _, s := range expectedServices {
+		if _, ok := services[s]; !ok {
+			t.Errorf("Expected service '%s' to be present", s)
+		}
+	}
+}
```

Comment:
As I said to Gabriel, skip testing `initTemplate` and `newResource` unless they're trying to do anything fancy. Testing if a function sets a couple of fields in a struct (that has a high probability of changing in the near future) doesn't provide any real value and has an increased risk of adding problems.